### DISCUSSION TITLE
.Net Maui Blazor Payment pages for windows

### DISCUSSION THREAD

**Discussion by 03V** [dotnet/maui#15847] [Q&A]
Our application uses an external payment provider to initiate payments. 

The scenario I am trying to achieve is :

Once the application receives the payment redirect URL, the app will redirect the user to a browser. Once the transaction is complete, the payment page will attempt to reopen the the paused instance of the application by using a deep link e.g. testapp://testcomponet/onsuccess

Unfortunately there does not seem to be an easy way to do this for a windows application. 

I am hoping for any help or to even know if this is possible with the current state of blazor maui as we may need to pivot to another tech stack that can support the mentioned scenario 

The closet I have gotten is by adding the following override in App.xamal.cs, but there does not seem to be way to reach the correct razor page

```c#
protected override void OnLaunched(LaunchActivatedEventArgs args)
    {
        var eventargs = Microsoft.Windows.AppLifecycle.AppInstance.GetCurrent().GetActivatedEventArgs();
        if (eventargs.Kind is ExtendedActivationKind.Protocol && eventargs.Data is Windows.ApplicationModel.Activation.ProtocolActivatedEventArgs activationArgs)
        {

            var uri = activationArgs.Uri;
            var authority = uri.Authority;
            var localPath= uri.LocalPath;

        }
        base.OnLaunched(args);
    }

``` 

Thanks in advanced

**Comment by Mistofs** [comment]
Interesting, so the application opens the browser on user's device to do the transaction, then comes back to the payment page which was paused until payment was complete?

**Reply by 03V** [reply]
That is correct @Mistofs  , the workaround I have is that after redirecting the customer to the browser, the application is terminated. The application is then re-opened once the transaction is completed on the payment provider page using the deep link. 
However, this is not the best user experience as the user may close the browser.
What would be best is if the application stays open, and once the transaction is complete, the running application is notified (but if it is not running, then open a new instance), and the application can direct the user to a summary page
